File: src/copilot_money_mcp/models/category.py
"""
Category model for Copilot Money data.
"""

from typing import Optional

from pydantic import BaseModel


class Category(BaseModel):
    """
    Represents a spending category from Copilot Money.

    Categories can be hierarchical with parent-child relationships.
    """

    model_config = {"strict": True, "populate_by_name": True}

    # Required fields
    category_id: str
    name: str

    # Optional fields
    parent_category_id: Optional[str] = None
    icon: Optional[str] = None
    color: Optional[str] = None

File: src/models/category.ts
/**
 * Category model for Copilot Money data.
 *
 * Categories can be hierarchical with parent-child relationships.
 */

import { z } from 'zod';

/**
 * Category schema with validation.
 */
export const CategorySchema = z
  .object({
    // Required fields
    category_id: z.string(),
    name: z.string(),

    // Optional fields
    parent_category_id: z.string().optional(),
    icon: z.string().optional(),
    color: z.string().optional(),
  })
  .strict();

export type Category = z.infer<typeof CategorySchema>;
